### PR TITLE
Phone field documentation

### PR DESCRIPTION
**Pull request by jorgetwgroup**
Documentation to work with new phone field in https://github.com/Laravel-Backpack/PRO/pull/103

### CODE REVIEW

File: 5.x/crud-fields.md [hunk 1]
```diff
@@ -1635,6 +1635,43 @@ Input preview:
...
+```php
+[   // phone
+    'name'  => 'phone', // db column for phone
+    'label' => 'Phone',
+    'type'  => 'phone',
+
+    // OPTIONALS
+    // options offically supported and tested, this options will be prioritized to config parameters
+    // this mean if you send parameter "initialCountry" => "us" 
+    // and "config" => ["initialCountry" => "us"], "cl" value will be used
```
Comment (tabacitu) [inline 5.x/crud-fields.md:1654]: I'd move this note AFTER the "officially supported" options. They only need to know about that if they wanna go deeper, so that's a better moment to say it I think, and we introduce them to `config` then.

So I'd rephrase to:

_the above are some officially supported and tested options, provided by intlTelInput.js; there are more, we just haven't tested them; check out their docs - you can use any of them inside the config attribute; take note that the directly defined attributes will override anything defined in the config;_

File: 5.x/crud-fields.md [hunk 2]
```diff
@@ -1635,6 +1635,43 @@ Input preview:
...
+        'onlyCountries' => ['pt', 'ro'],
+    ]
+],
+```
+
+Please note it is recommended that you check [official documentation](https://github.com/jackocnr/intl-tel-input) to get parameters description.
+
+Your end result will look like this:
+
+Input preview:
```
Comment (tabacitu) [inline 5.x/crud-fields.md:1671]: These two lines say the same thing, so let's just choose one of them.

File: 5.x/crud-fields.md [hunk 3]
```diff
@@ -1635,6 +1635,43 @@ Input preview:
...
+
+![CRUD Field - phone](https://user-images.githubusercontent.com/1032474/204588174-48935030-54e6-4a30-b34c-7e94220ae242.png)
+
```
Comment (tabacitu) [inline 5.x/crud-fields.md:1674]: I think it'd be good to also introduce people to the Phone Validation package here. Just like a sidenote, something like "you can validate this using Laravel's default `numeric` or if you want something advanced, we recommend _link_"

**Review by tabacitu — CHANGES_REQUESTED** [review]

File: 5.x/crud-fields.md [hunk 4]
```diff
@@ -1635,6 +1635,41 @@ Input preview:
 
 > NOTE: if you are having trouble uploading big images, please check your php extensions **apcu** and/or **opcache**, users have reported some issues with these extensions when trying to upload very big images. REFS: https://github.com/Laravel-Backpack/CRUD/issues/3457
 
+<hr>
+
+<a name="phone"></a>
+### phone <span class="badge badge-pill badge-info">PRO</span>
+
+Show a telephone input and let the user choose a flag from dropdown.
```
Comment (tabacitu) [inline 5.x/crud-fields.md:1643]:
```suggestion
Show a telephone number input. Lets the user choose the prefix using a flag from dropdown.
```

File: 5.x/crud-fields.md [hunk 5]
```diff
@@ -1635,6 +1635,41 @@ Input preview:
...
+
+    // OPTIONALS
+    'initialCountry' => 'cl',
+    'separateDialCode' => true,
+    'nationalMode' => true,
+    'autoHideDialCode' => false,
+    'placeholderNumberType' => 'MOBILE',
+    // the above are some officially supported and tested options, provided by intlTelInput.js;
+    // there are more, we just haven't tested them; check out their docs - you can use any of them 
+    // inside the config attribute; take note that the directly defined attributes will override anything defined in the config;
```
Comment (tabacitu) [inline 5.x/crud-fields.md:1659]: I've just suggested changing the merge order in https://github.com/Laravel-Backpack/PRO/pull/103/files so this could become:

```suggestion
    // inside the config attribute; take note that attributes defined in `config` will override any defaults or previously set values;
    ```

File: 5.x/crud-fields.md [hunk 6]
```diff
@@ -1635,6 +1635,41 @@ Input preview:
...
+    'config' => [
+        'onlyCountries' => ['pt', 'ro'],
+    ]
+],
+```
+
+Please note it is recommended that you check [official documentation](https://github.com/jackocnr/intl-tel-input) to get parameters description.
```
Comment (tabacitu) [inline 5.x/crud-fields.md:1666]:
```suggestion
For more info about parameters please see this JS plugin's [official documentation](https://github.com/jackocnr/intl-tel-input).
```

**Review by tabacitu — CHANGES_REQUESTED** [review]

**Comment by tabacitu** [comment]
Here's also a pic for the article:

Xnapper-2022-11-29-17 29 54

File: 5.x/crud-fields.md [hunk 7]
```diff
@@ -1635,6 +1635,42 @@ Input preview:
...
+
+    // OPTIONALS
+    'initialCountry' => 'cl',
+    'separateDialCode' => true,
+    'nationalMode' => true,
+    'autoHideDialCode' => false,
+    'placeholderNumberType' => 'MOBILE',
+    // the above are some officially supported and tested options, provided by intlTelInput.js;
+    // there are more, we just haven't tested them; check out their docs - you can use any of them 
+    // inside the config attribute; take note that attributes defined in `config` will override any defaults or previously set values;
```
Comment (pxpm) [inline 5.x/crud-fields.md:1659]:
```suggestion
    // most options provided by intlTelInput.js are supported, you can try them out using the `config` attribute;
    //  take note that options defined in `config` will override any default values from the field;
```

File: 5.x/crud-fields.md [hunk 8]
```diff
@@ -1635,6 +1635,42 @@ Input preview:
...
+
+Show a telephone number input. Lets the user choose the prefix using a flag from dropdown.
+
+```php
+[   // phone
+    'name'  => 'phone', // db column for phone
+    'label' => 'Phone',
+    'type'  => 'phone',
+
+    // OPTIONALS
```
Comment (pxpm) [inline 5.x/crud-fields.md:1651]:
```suggestion
    //  DEFAULTS
```

File: 5.x/crud-fields.md [hunk 9]
```diff
@@ -1635,6 +1635,41 @@ Input preview:
...
+    'nationalMode' => true,
+    'autoHideDialCode' => false,
+    'placeholderNumberType' => 'MOBILE',
+    // most options provided by intlTelInput.js are supported, you can try them out using the `config` attribute;
+    //  take note that options defined in `config` will override any default values from the field;
+    ```
+    'config' => [
+        'onlyCountries' => ['pt', 'ro'],
+    ]
+],
```
Comment (tabacitu) [inline 5.x/crud-fields.md:1663]:
```suggestion
    // most options provided by intlTelInput.js are supported, you can try them out using the `config` attribute;
    //  take note that options defined in `config` will override any default values from the field;
    'config' => [
        'onlyCountries' => ['pt', 'ro'],
        'initialCountry' => 'cl',
        'separateDialCode' => true,
        'nationalMode' => true,
        'autoHideDialCode' => false,
        'placeholderNumberType' => 'MOBILE',
    ]
],
```
